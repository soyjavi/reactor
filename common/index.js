import { LAYOUT } from './layout/index';
import SHAPE from './shape';
import THEME from './theme';

export { LAYOUT, SHAPE, THEME };
