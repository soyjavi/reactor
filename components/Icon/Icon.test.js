import React from 'react';
import renderer from 'react-test-renderer';

import { THEME } from '../../common';
import { Icon } from '..';
import MOCKS from './Icon.mocks';

jest.useFakeTimers();

describe('<Icon>', () => {
  it('renders', () => {
    const tree = renderer.create(<Icon />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {value}', () => {
    const tree = renderer.create(<Icon value={MOCKS.VALUE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {color} (only Native)', () => {
    const tree = renderer.create(<Icon color={MOCKS.COLOR} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {size}', () => {
    const tree = renderer.create(<Icon size={MOCKS.SIZE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {family}', () => {
    const tree = renderer.create(<Icon value={MOCKS.VALUE} family={MOCKS.FAMILY} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
