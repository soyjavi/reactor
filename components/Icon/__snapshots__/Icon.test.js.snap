// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon> renders 1`] = `<Text />`;

exports[`<Icon> when {color} (only Native) 1`] = `<Text />`;

exports[`<Icon> when {family} 1`] = `<Text />`;

exports[`<Icon> when {size} 1`] = `<Text />`;

exports[`<Icon> when {value} 1`] = `<Text />`;
